nwb_conversion_tools/nwbconverter.py

```diff
@@ -60,25 +61,27 @@ def get_metadata(self):
             metadata = dict_deep_update(metadata, interface_metadata)
         return metadata
 
-    def run_conversion(self, metadata_dict, nwbfile_path=None, save_to_file=True, stub_test=False):
+    def run_conversion(self, metadata_dict, nwbfile_path=None, save_to_file=True, **conversion_options):
         """Build nwbfile object, auto-populate with minimal values if missing."""
         nwbfile_kwargs = dict(
-                session_description="no description",
-                identifier=str(uuid.uuid4()),
-                session_start_time=datetime.now()
-            )
-
+            identifier=str(uuid.uuid4()),
+            session_start_time=datetime.now()
+        )
         if 'NWBFile' in metadata_dict:
             nwbfile_kwargs.update(metadata_dict['NWBFile'])
-
         if 'Subject' in metadata_dict:
             nwbfile_kwargs.update(subject=Subject(**metadata_dict['Subject']))
-
         nwbfile = NWBFile(**nwbfile_kwargs)
 
-        # Run data interfaces data conversion
+        conversion_data_routed = dict()
+        for interface_name, interface in self.data_interface_classes.items():
+            conversion_schema = interface.get_conversion_options_schema()
+            conversion_data_routed[interface_name] = get_schema_data(
+                conversion_options[interface_name],
+                conversion_schema
+            )
         for name, data_interface in self.data_interface_objects.items():
-            data_interface.convert_data(nwbfile, metadata_dict, stub_test)
+            data_interface.convert_data(nwbfile, metadata_dict, **conversion_data_routed[name])
```

Comment:
Can you combine these loops?

Reply:
Combined the occurrence of these loops here, as well as the similar structure in the input arguments